packages/nodes-base/nodes/Postgres/v1/PostgresV1.node.ts

```diff
@@ -298,33 +298,27 @@ export class PostgresV1 implements INodeType {
 				this: ICredentialTestFunctions,
 				credential: ICredentialsDecrypted,
 			): Promise<INodeCredentialTestResult> {
-				const credentials = credential.data as IDataObject;
-				try {
-					const pgp = pgPromise();
-					const config: IDataObject = {
-						host: credentials.host as string,
-						port: credentials.port as number,
-						database: credentials.database as string,
-						user: credentials.user as string,
-						password: credentials.password as string,
-					};
+				const credentials = credential.data as PostgresNodeCredentials;
 
-					if (credentials.allowUnauthorizedCerts === true) {
-						config.ssl = {
-							rejectUnauthorized: false,
-						};
-					} else {
-						config.ssl = !['disable', undefined].includes(credentials.ssl as string | undefined);
-						config.sslmode = (credentials.ssl as string) || 'disable';
-					}
+				let connection: PgpConnection | undefined;
 
-					const db = pgp(config);
-					await db.connect();
```

Comment:
This would have left the connection hanging. It needs to be ended:

https://vitaly-t.github.io/pg-promise/Database.html#connect